tests/infrastructure/test_data_manager_stack.py

```diff
@@ -26,3 +32,824 @@ def test_sds_data_manager_stack(app, sds_id):
+                                    "Arn",
+                                ]
+                            },
+                            {
+                                "Fn::Join": [
+                                    "",
+                                    [
+                                        {
+                                            "Fn::GetAtt": [
+                                                Match.string_like_regexp("DATABUCKET*"),
+                                                "Arn",
+                                            ]
+                                        },
+                                        "/*",
+                                    ],
+                                ]
+                            },
+                        ],
+                    }
+                ],
+            },
+        },
+    )
+
+
+def test_custom_s3_auto_delete_resource_count(template):
+    template.resource_count_is("Custom::S3AutoDeleteObjects", 1)
+
+    template.has_resource_properties(
+        "Custom::S3AutoDeleteObjects",
+        {
+            "ServiceToken": {
+                "Fn::GetAtt": [
+                    Match.string_like_regexp(
+                        "CustomS3AutoDeleteObjectsCustomResourceProviderHandler*"
+                    ),
+                    "Arn",
+                ]
+            },
+            "BucketName": {"Ref": Match.string_like_regexp("DATABUCKET*")},
+        },
+    )
+
+
+def test_custom_s3_bucket_notifications_resource_properties(template):
+    template.resource_count_is("Custom::S3BucketNotifications", 1)
+
+    template.has_resource_properties(
+        "Custom::S3BucketNotifications",
+        {
+            "ServiceToken": {
+                "Fn::GetAtt": [
+                    Match.string_like_regexp("BucketNotificationsHandler*"),
+                    "Arn",
+                ]
+            },
+            "BucketName": {"Ref": Match.string_like_regexp("DATABUCKET*")},
+            "NotificationConfiguration": {
+                "LambdaFunctionConfigurations": [
+                    {
+                        "Events": ["s3:ObjectCreated:*"],
+                        "LambdaFunctionArn": {
+                            "Fn::GetAtt": [
+                                Match.string_like_regexp("IndexerLambda*"),
+                                "Arn",
+                            ]
+                        },
+                    }
+                ]
+            },
+            "Managed": True,
+        },
+    )
+
+
+def test_secrets_manager_resource_count(template):
+    template.resource_count_is("AWS::SecretsManager::Secret", 1)
+
+
+def test_secrets_manager_resource_properties(template):
+    template.has_resource(
+        "AWS::SecretsManager::Secret",
+        {
+            "DeletionPolicy": "Delete",
+            "UpdateReplacePolicy": "Delete",
+        },
+    )
+
+
+def test_opensearch_domain_resource_count(template):
```

Comment:
I guess this is a personal preference for how much to put into each individual test. I probably would have went for `def test_opensearch_domain(template, sds_id)` and combined the count and properties into a single test... This is fine though and should give more informative failure messages.

Reply:
Sounds good. I think it's good practice to have one test per function and to have very descriptive names so it's clear what the issue is if one fails.